crates/rolldown/tests/rolldown/topics/new_url/dataurl/main.js

```diff
@@ -0,0 +1,4 @@
98QfaViNwFMuESXTD0c6IathoZd9juDGeJbekEDD4soStX0VVb6Fq36Dov6J6n6CZzdHP+71+gm0/QTV7WSFQvnTWwOy10rSq6ahldKUQXOEtsS8e4w3ZnUA1bnegc2wWRK5AtXCL6iuwrEVEFWeUfX+Ba+P9QG1ebVh7jOlq11UfV31pFJVQFWmt9YHq31lExFamOjFWgDc1eGvr2zE1oT+KRe7T5JZ5LiumN0Iy9OHQmschTim00sTSNeGR3kXRtSjIzxDHMUrLBDMUspKQjgUFElHzuE8FnntLMC/1EjtKNZrrNIko7HOo1DCj9bFenboZUtJtSF9m0SU3jdKTH6MwgZb1LqZ689EjpvZ5qvT1S3Sy8q/RRNElDq9BXpV+ySNOto7ZIQMYR7eMt0tmvfF3E2Vx8rfikv3tRu1smsbyrnbuE0nCCUrV1+zKdM88nL51WtRw6BqkJVlA4IGQHAAAwIgCdASpgAGAAPm0skUWkIqGYCx6oQAbEtgBa6tvsjx4+8/kz7Itbfs34U5d+ZvNx5d/6fSA8wD9OelP5gP2P/cj3wPSB6AH8y/5/WEegB5bHskfuZ+6ntQ5sH/WesVXp78+v5vScCeCW8IQx5/r/cB7d/ob/d+4L/Jf6l6N3sM9Ar9e3ErDkoJ/9UifC+fXJ9nIbcjMv/uSILosFlG3Az3onvqN2sJBHC51QxoxWOdeYXJ/5/f0lElFjVcvNNiZE+YebqVrnd3zofCPPNjz3Em2bo08rM7eCHMHCJsS4mdSvUr9yj/JJF/ty1lZdEiv0iWddODofUG7kq3S/mzrmagJPLYjBATh7b9leoenvYLEi05utl5w3aoAA/v6eb//koTX+wieRQ1dv/92WN/sefyI/8Tlw5RntMaqB6PsZo+Vz8WxHT2MnAqNscsDjQyI+Sg1bD8+Kb03vgS4W8+eB7IxS2Mvn6PuCdK0VtjA7/R4GEvvXp7HsdmmSSfOjxpS4fc/dvKzr5JL+SKL96CchxHDdABrr42gQF90Sy0SjEsTqwcO3oypKRmmZDCl9RQNJ4RsE9vWhqh/jLlUD6yelsEG6Kj52XLe3qm/fRg16oFwv0kFom/HByDIc5czWPqRuEwC5AN8H1gHs3rDaQEyUisnoNwLWSPhFuXo/AqoOyvFwjR+CceNTvsu2roqH5yI2AvHS/G1m8QdYH2+0EzM+znCR3A4C/VzfBlfRHRZ9MRXS390ZWZICmElAMGUN6elL9+R72fUVgePA/RQinUmKTi7xhTQgvrPahLm9JFItK5bZ4t9XuQX1+hXi4KY+RcL8IPWNqnOEkS4ySJcc0Hj/0cA3CPERn4DdjTNgG4MG7E6NLDY8ABVoMzaHDQ4LuHMZ1H/UhtuLnE62jMYVHO5AjP69GUaONvGYYoSMkD/cvVy/xTHtjFn2RYk4YsCGJD/1GTJ42pdqJWaIeCSrpdopYnn8lZ3nfQQUaVjRECRe2E6at5cYLxZEPZHcaZhB6j/0CYKdX0c1fI6aLFHEpWyjfPOS9BKj/DuKvg8ZBXphv+MgrsfK4zDx52gTfQSs6sFavIDrkmgy1Pmu3CEgJpMOvTaRSacksSHBmI2l2S/xM+v6MevFoK7NRQ941Ueiqk4JFTugODS+10kRsTpO0/5FG/3VkzOkBsRdwQ9/g1JLLMpd8Pm46RDKTwRi3DZGGIjc5VZh2rn2iUshHbpPh7JqTqSpbrzh5Q71pSR1NpCuqpmL3BtHrad2Zz1DM1CLETrSGu5F85x2XR6mAigcbccHfLhBx3iHTR7xz8C01KlMMOBumXFbM/kEc3GLyOT+apHX21Sx/t9/rm92U7HrwSNd+/C797kzj1g4jK7jH2SW+qo0DTiB18Wj2GMomi5S/YBxoaETWsKBIsfXispyT5H1N6MZtijmNaI0YO7/epZeiRdHh1aac7cyypOHKN3Ttpeq5G+gO3iW4HS73N0nMDLYsAwuecSLL4YxCYqLVIw3mcDZWx4+mQjSnpAQdl5ib8WbN/nmJ+PXdPE//8iO1eA82MllY1VxldDEBzOXsfywJCkRVYPEbWBPn6H9KCEoGvHe48rpHaFJiDmk1BFUXJJHolFvjAyJnhyNp+CSBE4JtCfqC/7sccZB7Nv6DxpxReXe5mFezagJZIa8VyQ462I8Po3LPGyqShObodpyJI3cWKqiDJ3mzFJiOu+Kosyi+lr/V9jjWvlhCkf9uXa1QL5f3OHwN8P+z67YPwDRmTjc8j2oJ1ECn8w70f+XlFyfJrlGOpPjS93jrAWgN+X/+uUcuKZYZE3EQCRsFTJ42fqTZ5eCt/lXLrdjudA3vLexyBZzsyAJOIyu7uiv6olL/5pB+5YeakMelfC+rAL4NYGFiXaWO7eI1ID7fteXqr8y3zFn08xT4HdC6DjRb9bFX/IAkyWdzGAu1bpTVC02BVoJFa0PoGboq42Hwtpq1Yvp6LiIb5NU9LJ4EYRFHwEeOKLHas0PUN8UeiR+3zRTGCK38+ihBF4IL8IWvjQspsbbOperpfCxOq7tgLljv+Dj5ZrtPWolBGkY+YAj0KGuLYPXphkmZ3he7+UY9iGvgVNmHwMqrJs9RhAHrsbmBNdF/OD9tDFTyY/eroy6BYmBP1sd0BxMpQCO0wipCQB07biLfiz0iGQh/j5pbbiooso3ysDKgU6iEAENZ7GQtgcTxDAhPN5g/Y2OvfstKWAzeNAcgrfAfHreS9Cfa6M8NOpboCUg0xhYa7vBdl5VguSL4nHoDYDiztuQGrx4OgYPAtRFiVfWhdOCWLSXZ4ljQGi+KW3rG5V0X01dYZz6evIXP9wqHhkekw2K4y842vn82Rb6t+Pi6pgups0s1uNfLhui1LRbW34my38Go3JeuNaxQAfyW+4LPx9LvOHRxLR8XmMi3TGw1pSsP0Je64S2R4lD6XxOa/ujg+hf3Z3//R///o4r//ovn/9unf+Ebnn8cZgMBAwsVra6s78StHcigPdOIHNGzgAAAA==',
```

Comment:
Do we really need such complex `data URL` as a snapshot test case?

Reply:
Need real-world data to check if everthing work as expected. Sush as logic here https://github.com/rolldown/rolldown/blob/2b4f224f665006daaccc8755847cc17607efa33a/crates/rolldown_plugin_data_url/src/data_url_plugin.rs#L39.

I didn't find a small dataurl data. This is copied from other tests.

Reply:
The current solution may changes in the future. Need real-world data to check if everything work as expected even in the future changes.

Reply:
We could switch to a small real-world dataurl I guess.